src/components/Tooltip.js

```diff
@@ -0,0 +1,183 @@
+import _ from 'underscore';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Animated, Text, View} from 'react-native';
+import Hoverable from './Hoverable';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import getTooltipStyles from '../styles/getTooltipStyles';
+
+const propTypes = {
+    // The text to display in the tooltip.
+    text: PropTypes.string.isRequired,
+
+    // Children to wrap with Tooltip.
+    children: PropTypes.node.isRequired,
+
+    // Props inherited from withWindowDimensions
+    ...windowDimensionsPropTypes,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // The distance between the left side of the wrapper view and the left side of the window
+            xOffset: 0,
+
+            // The distance between the top of the wrapper view and the top of the window
+            yOffset: 0,
+
+            // The width and height of the wrapper view
+            wrapperWidth: 0,
+            wrapperHeight: 0,
+
+            // The width and height of the tooltip itself
+            tooltipWidth: 0,
+            tooltipHeight: 0,
+        };
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+
+        this.animation = new Animated.Value(0);
+
+        this.getWrapperPosition = this.getWrapperPosition.bind(this);
+        this.measureWrapperAndGetPosition = this.measureWrapperAndGetPosition.bind(this);
+        this.measureTooltip = this.measureTooltip.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        /*
+           We only want to re-render if props or state have changed.
+           This prevents an infinite rendering loop with the `onLayout` callback calling setState.
+           Also, we need to do a deep comparison of props using _.isEqual rather than relying on the shallow comparison
+           done by `PureComponent` because a shallow comparison would not catch changes in `props.windowDimensions`.
+         */
+        return !(_.isEqual(this.props, nextProps) && _.isEqual(this.state, nextState));
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(this.props.windowDimensions, prevProps.windowDimensions)) {
+            this.getWrapperPosition()
+                .then(({x, y}) => {
+                    if (x !== this.state.xOffset || y !== this.state.yOffset) {
```

Comment:
I think this check is no longer needed. We won't re-render if these state value have not changed so we should be good.